### DISCUSSION TITLE
How to find the order of a ordinary/partial differential equation in sympy

### DISCUSSION THREAD

**Discussion by ShrohanMohapatra** [sympy/sympy#25242] [General]
I am looking for an API/strategy to use one/some attributes of the relevant classes to find the order of an ODE/PDE using Sympy. Can anyone help, please?

**Comment by oscarbenjamin** [comment]
I've converted this to a discussion because issues should not be used for getting help.

**Reply by ShrohanMohapatra** [reply]
Sounds good .... so any suggestions on how to find the order of a ODE/PDE? Does the result of dsolve or pdsolve return any object which records the order?

**Comment by Arniiiii** [comment]
> I am looking for an API/strategy to use one/some attributes of the relevant classes to find the order of an ODE/PDE using Sympy. Can anyone help, please?

https://github.com/sympy/sympy/blob/fcf489588fa3ff1c06836719ece17b71e9839108/sympy/solvers/deutils.py#LL94C26-L94C26
if this doesn't help, check internal fucntions of collect functions:
https://github.com/sympy/sympy/blob/fcf489588fa3ff1c06836719ece17b71e9839108/sympy/simplify/radsimp.py#L235
for amazing experience use parse_term too: https://github.com/sympy/sympy/blob/fcf489588fa3ff1c06836719ece17b71e9839108/sympy/simplify/radsimp.py#L262

**Comment by oscarbenjamin** [comment]
There is a function called `ode_order`:
```
In [6]: ode_order(f(x).diff(x, 2) + f(x), f(x))
Out[6]: 2
```
